stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/call/utils/SoundInputProcessor.kt

```diff
@@ -54,26 +62,24 @@ internal class DecibelThresholdDetection(
         }
     }
 
-    private fun calculateAverageDecibelPower(pcmData: ByteArray): Double {
+    private fun calculateAverageDecibelPower(soundData: ByteArray): Double {
         // For code clarity the following code expects the pcmData to be in PCM 16bit format.
         // And this is the default format used by the JavaAudioDeviceModule, so unless someone
         // in this class will override the default value then we are safe.
+        val samples = ShortArray(soundData.size / 2) { i ->
+            ((soundData[i * 2 + 1].toInt() and 0xFF) shl 8 or (soundData[i * 2].toInt() and 0xFF)).toShort()
+        }
 
-        val referencePressure = 0.00002 // 20 µPa (reference pressure for 0 dB)
```

Comment:
(I used this opportunity here to simplify my original code for decibel calculations)